src/staticanalysis/bot/static_analysis_bot/workflows/local.py

```diff
@@ -55,17 +57,31 @@ def __init__(self, parent, analyzers, index_service):
         self.index_service = index_service
 
     @stats.api.timed('runtime.clone')
-    def clone(self):
+    def clone(self, revision):
         '''
         Clone mozilla-unified
         '''
         logger.info('Clone mozilla unified', dir=settings.repo_dir)
-        cmd = hglib.util.cmdbuilder('robustcheckout',
-                                    REPO_UNIFIED,
-                                    settings.repo_dir,
-                                    purge=True,
-                                    sharebase=settings.repo_shared_dir,
-                                    branch=b'central')
+
+        if settings.source == SOURCE_TRY:
+            # Clone Try using the target revision
+            assert revision.mercurial_revision is not None, 'Missing try mercurial revision'
+            cmd = hglib.util.cmdbuilder('robustcheckout',
+                                        REPO_TRY,
+                                        settings.repo_dir,
+                                        revision=revision.mercurial_revision,
+                                        upstream=REPO_UNIFIED,
```

Comment:
I wonder if we can't clone try without using unified, it should be faster (as we wouldn't need to get commits from branches we don't need).

Reply:
I reused the same arguments as on MC from `run-task`, without the fingerprinting:

```
hg robustcheckout \
	--sharebase  /builds/worker/checkouts/hg-store \
	--purge \
	--upstream https://hg.mozilla.org/mozilla-unified \
	--revision XXXXX \
	https://hg.mozilla.org/try \
	/builds/worker/checkouts/gecko
```

Reply:
Yeah, but maybe if we don't use unified it will be faster. We can check later on (maybe file a follow-up about investigating the fastest way to clone try).